sdk/identity/Azure.Identity/src/DeviceCodeCredential.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Azure.Core;
+using Azure.Core.Pipeline;
+using Microsoft.Identity.Client;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Details of the device code to present to a user to allow them to authenticate through the device code authentication flow.
+    /// </summary>
+    public struct DeviceCode
+    {
+        /// <summary>
+        /// The UTC time when the device code request will expire.
+        /// </summary>
+        public DateTimeOffset ExpiresOn { get; set; }
+
+        /// <summary>
+        /// The message to display to the user to allow them to authenticate through the device code authentication flow.
+        /// </summary>
+        public string Message { get; set; }
```

Comment:
I see that you are not including the user code and the login url in here - are we making the assumption that DeviceCode will not be used in any automations, but only through human interaction?

Reply:
I included them in mine in case someone wants to show the code in a specific way or send the URI to another device to be loaded up.